Source/Diagnostics/ReducedDiags/DifferentialLuminosity.cpp

```diff
@@ -218,34 +217,57 @@ void DifferentialLuminosity::ComputeDiags (int step)
                         index_type const j_1 = indices_1[i_1];
                         index_type const j_2 = indices_2[i_2];
 
-                        Real const u1_square =  u1x[j_1]*u1x[j_1] + u1y[j_1]*u1y[j_1] + u1z[j_1]*u1z[j_1];
-                        Real const gamma1 = std::sqrt(1._rt + u1_square*inv_c2);
-                        Real const u2_square = u2x[j_2]*u2x[j_2] + u2y[j_2]*u2y[j_2] + u2z[j_2]*u2z[j_2];
-                        Real const gamma2 = std::sqrt(1._rt + u2_square*inv_c2);
-                        Real const u1_dot_u2 = u1x[j_1]*u2x[j_2] + u1y[j_1]*u2y[j_2] + u1z[j_1]*u2z[j_2];
+                        Real p1t=0, p1x=0, p1y=0, p1z=0; // components of 4-momentum of particle 1
+                        Real const u1_sq =  u1x[j_1]*u1x[j_1] + u1y[j_1]*u1y[j_1] + u1z[j_1]*u1z[j_1];
+                        if (m1 != 0) {
+                            p1t = m1*std::sqrt( c_sq + u1_sq );
+                            p1x = m1*u1x[j_1];
+                            p1y = m1*u1y[j_1];
+                            p1z = m1*u1z[j_1];
+                        } else { // photon case (momentum is normalized by m_e)
+                            p1t = PhysConst::m_e*std::sqrt( u1_sq );
+                            p1x = PhysConst::m_e*u1x[j_1];
+                            p1y = PhysConst::m_e*u1y[j_1];
+                            p1z = PhysConst::m_e*u1z[j_1];
+                        }
```

Comment:
Would this improve the code?
No need of an explicit if-else statement but there a few more multiplications. 
 
```
amrex::Real const mass = (m1 != 0) ? m1 : PhysConst::m_e;
p1t = std::sqrt( m1*m1*c_sq + mass*mass*u1_sq )
p1x = mass * u1x[j_1];
p1y = mass * u1y[j_1];
p1z = mass * u1z[j_1];
```